libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/internal/MapboxTripNotification.kt

```diff
@@ -314,125 +315,76 @@ class MapboxTripNotification constructor(
         notificationView.updateArrivalTime(time)
     }
 
-    private fun updateCurrentManeuverToDefault(isFreeDriveMode: Boolean) {
-        if (isFreeDriveMode) {
-            currentManeuverType = null
-            currentManeuverModifier = null
-            currentRoundaboutAngle = null
-        }
+    private fun isInstructionTextChanged(primaryText: String): Boolean {
+        return currentInstructionText.isNullOrEmpty() || currentInstructionText != primaryText
     }
 
-    private fun updateManeuverImage(drivingSide: String) {
-        getManeuverBitmap(
-            currentManeuverType ?: "",
-            currentManeuverModifier,
-            drivingSide,
-            currentRoundaboutAngle
-        )?.let { bitmap ->
-            notificationView.updateImage(bitmap)
-        }
+    private fun updateInstructionText(primaryText: String) {
+        notificationView.updateInstructionText(primaryText)
+        currentInstructionText = primaryText
     }
 
     private fun isManeuverStateChanged(bannerInstruction: BannerInstructions): Boolean {
         val previousManeuverType = currentManeuverType
         val previousManeuverModifier = currentManeuverModifier
         val previousRoundaboutAngle = currentRoundaboutAngle
 
-        updateManeuverState(bannerInstruction)
+        currentManeuverType = bannerInstruction.primary().type()
+        currentManeuverModifier = bannerInstruction.primary().modifier()
+        currentRoundaboutAngle = bannerInstruction.primary().degrees()?.toFloat()
 
         return !TextUtils.equals(currentManeuverType, previousManeuverType) ||
             !TextUtils.equals(currentManeuverModifier, previousManeuverModifier) ||
             currentRoundaboutAngle != previousRoundaboutAngle
     }
 
-    private fun updateManeuverState(bannerInstruction: BannerInstructions) {
-        currentManeuverType = bannerInstruction.primary().type()
-        currentManeuverModifier = bannerInstruction.primary().modifier()
-
-        currentRoundaboutAngle = if (ROUNDABOUT_MANEUVER_TYPES.contains(currentManeuverType)) {
-            bannerInstruction.primary().degrees()?.toFloat()?.let {
-                adjustRoundaboutAngle(it)
+    private fun updateManeuverImage(drivingSide: String) {
+        val notificationTurnIcon = turnIconHelper.retrieveTurnIcon(
+            currentManeuverType,
+            currentRoundaboutAngle,
+            currentManeuverModifier,
+            drivingSide
+        )
+        notificationTurnIcon?.let { turnIcon ->
+            ifNonNull(turnIcon.icon) { image ->
+                val originalDrawable = notificationView.getImageDrawable(image)
+                ifNonNull(originalDrawable) { drawable ->
+                    getManeuverBitmap(drawable, turnIcon.shouldFlipIcon)?.let { bitmap ->
+                        notificationView.updateImage(bitmap)
+                    }
+                }
```

Comment:
Wondering if it'd be clearer to use `ifNonNull` or `?.let {` consistently? 🤔

Reply:
sounds good 👍🏻